dot/network/notifications.go

```diff
@@ -321,7 +321,7 @@ func (s *Service) broadcastExcluding(info *notificationsProtocol, excluding peer
 }
 
 func (s *Service) readHandshake(stream libp2pnetwork.Stream, decoder HandshakeDecoder) <-chan *handshakeReader {
-	hsC := make(chan *handshakeReader)
+	hsC := make(chan *handshakeReader, 100)
```

Comment:
this should only read the handshake once, I think you can leave this as unbuffered

Reply:
Updated.